app/gui/src/project-view/components/visualizations/TableVisualization/tableVizToolbar.ts

```diff
@@ -38,23 +37,33 @@ export interface FormatMenuOptions {
   textFormatterSelected: Ref<TextFormatOptions>
 }
 
+export interface ColumnNodeButton {
+  isCreateNewNodeEnabled: ToValue<boolean>
+  createNodes: (...options: NodeCreationOptions[]) => void
+  hiddenColumns: Ref<string[]>
+  vizColumnOrder: Ref<string[]>
```

Comment:
This API could use some clarification--it's not obvious that `[]` is the way to indicate that the order has not been overridden; also, the use of an empty array raises the question of what happens in general if the array doesn't contain every input
 I think rather than expose the full `reorder_columns` semantics here, I'd constrain the interface more: Change the value to `string[] | undefined`, and document that the caller should include all columns. This gives the API user all the capabilities they need, without including the full complexity of the `reorder_columns` behaviour in this API.